cmd/podman/spec.go

```diff
@@ -210,10 +210,8 @@ func createConfigToOCISpec(config *createConfig) (*spec.Spec, error) {
 	if config.Hostname != "" {
 		g.AddProcessEnv("HOSTNAME", config.Hostname)
 	}
-
-	for _, sysctl := range config.Sysctl {
-		s := strings.SplitN(sysctl, "=", 2)
-		g.AddLinuxSysctl(s[0], s[1])
+	for sysctlKey, sysctlVal := range config.Sysctl {
+		g.AddLinuxSysctl(sysctlKey, sysctlVal)
```

Comment:
Are you verifying the syscalls as legal?  We can do that in a separate PR.

Reply:
Legal as in if they are key=value pairs? If so, yes it i being checked in create.go

Reply:
Only certain sysctls are valid in containers. Not sure which. Docker likely maintains a list.

Reply:
```
// ValidateSysctl validates a sysctl and returns it.
func ValidateSysctl(val string) (string, error) {
        validSysctlMap := map[string]bool{
                "kernel.msgmax":          true,
                "kernel.msgmnb":          true,
                "kernel.msgmni":          true,
                "kernel.sem":             true,
                "kernel.shmall":          true,
                "kernel.shmmax":          true,
                "kernel.shmmni":          true,
                "kernel.shm_rmid_forced": true,
        }
        validSysctlPrefixes := []string{
                "net.",
                "fs.mqueue.",
        }
        arr := strings.Split(val, "=")
        if len(arr) < 2 {
                return "", fmt.Errorf("sysctl '%s' is not whitelisted", val)
        }
        if validSysctlMap[arr[0]] {
                return val, nil
        }

        for _, vp := range validSysctlPrefixes {
                if strings.HasPrefix(arr[0], vp) {
                        return val, nil
                }
        }
        return "", fmt.Errorf("sysctl '%s' is not whitelisted", val)
}
```

Reply:
Added validation.